src/sys/socket/mod.rs

```diff
@@ -2215,7 +2216,7 @@ pub fn recvfrom<T: SockaddrLike>(
             &mut len as *mut socklen_t,
         ))? as usize;
 
-        Ok((ret, T::from_raw(addr.assume_init().as_ptr(), Some(len))))
+        Ok((ret, T::from_raw(addr.as_ptr().cast(), Some(len))))
```

Comment:
This change isn't technically necessary, both versions are equivalent.

But I think my version better expresses what is happening here. Before, it was

`MaybeUninit<S>` -> `S` -> `*const sockaddr` -> `Option<S>`.

Converting `S` to `Option<S>` makes no sense. Now it is

`MaybeUninit<S>` -> `*const S` -> `*const sockaddr` -> `Option<S>`.